proposals/lip-0051.md

```diff
@@ -769,14 +770,243 @@ if the above send function does not fail:
+                                   } for each lockedBalance in userEntry.lockedBalances]
+               } serialized using userStoreSchema.
+  ```
+
+* For each entry `supplyEntry` in `genesisBlockAssetObject.supplySubstore`, create an entry in the supply substore  with
+  ```python
+  storeKey = uint32be(supplyEntry.localID)
+  storeValue = {"totalSupply": supplyEntry.totalSupply} serialized using supplyStoreSchema.
+  ```
+
+* For each entry `escrowEntry` in `genesisBlockAssetObject.escrowSubstore`, create an entry in the escrow substore  with
+  ```python
+  storeKey = uint32be(escrowEntry.escrowChainID) || uint32be(escrowEntry.localID)
+  storeValue = {"amount": escrowEntry.amount} serialized using escrowStoreSchema.
+  ```
+
+* Create an entry in the available local ID substore with
+  ```python
+  storeKey = EMPTY_BYTES
+  storeValue = {"nextAvailableLocalID": genesisBlockAssetObject.availableLocalIDSubstore.nextAvailableLocalID} 
+               serialized using availableLocalIDStoreSchema.
+  ```
+
+* For each entry `terminatedChainID` in `genesisBlockAssetObject.terminatedEscrowSubstore`, create an entry in the terminated escrow substore with
+  ```python
+  storeKey = uint32be(terminatedChainID)
+  storeValue = {"escrowTerminated": True} serialized using terminatedEscrowSchema.
+  ```
+  
+Once the module store is initialized, its validity is attested via the two checks below.
+* Check that for each native token the total supply is correct. 
+  This can be done by checking that the function below returns `True`.
+  ```python
+  validateSupplyStoreEntries():
+      computedSupplies = {}
+      for (storeKey, storeValue) in user substore:
+          chainID = storeKey[20:24] # part of the key corresponding to the chain ID of the token
+          localID = storeKey[20:28] # part of the key corresponding to the local ID of the token
+          if chainID == CHAIN_ID_ALIAS_NATIVE:
+              computedSupplies[localID] += sum of availableBalance and all locked amounts of storeValue
+              
+      for (storeKey, storeValue) in escrow substore:
+          localID = storeKey[4:8] # part of the key corresponding to the local ID of the token
+          computedSupplies[localID] += storeValue.amount
+          
+      if computedSupplies[localID] >= 2^64 for any localID:
+          return False
+      
+      storedSupplies = {}
+      for (storeKey, storeValue) in supply substore:
+          localID = storeKey
+          storedSupplies[localID] = storeValue.amount
+      
+      # check if both dictionary are coherent
+      for localID in computedSupplies:
+          if storedSupplies[localID] does not exist or computedSupplies[localID] != storedSupplies[localID]:
+              return False
+              
+      for localID in storedSupplies:
+          if (computedSupplies[localID] does not exist and storedSupplies[localID] != 0)
+              or (computedSupplies[localID] != storedSupplies[localID]):
```

Comment:
I think the line above still should to be removed because the condition will be evaluated for
`computedSupplies[localID]` not existing and `storedSupplies[localID]==0`. Note that also it is redundant as the relevant check is already done in the for loop above.

Reply:
Indeed, removed.